Mage/src/main/java/mage/util/CardUtil.java

```diff
@@ -1123,6 +1124,62 @@ public static boolean checkTargetedEventAlreadyUsed(String checkingReference, St
         return false;
     }
 
+    /**
+     * For overriding `canTarget()` with usages such as "any number of target cards with total mana value X or less".
+     * Call this after super.canTarget() returns true.
+     *
+     * @param selectedTargets this.getTargets()
+     * @param checkTargetId   id from canTarget
+     * @param valueMapper     e.g. MageObject::getManaValue or m -> m.getPower().getValue()
+     * @param maxValue        the maximum total value of the parameter
+     * @return true if the total value would not be exceeded by the target being checked.
+     */
+    public static boolean checkCanTargetTotalValueLimit(Collection<UUID> selectedTargets, UUID checkTargetId,
+                                                        ToIntFunction<MageObject> valueMapper, int maxValue,
+                                                        Game game) {
+        MageObject checkTarget = game.getObject(checkTargetId);
+        if (checkTarget == null) {
+            return false;
+        }
+        return maxValue >= selectedTargets.stream()
+                .map(game::getObject)
+                .filter(Objects::nonNull)
+                .mapToInt(valueMapper)
+                .sum()
+                + (selectedTargets.contains(checkTargetId) ? 0 : valueMapper.applyAsInt(checkTarget));
+    }
+
+    /**
+     * For overriding `possibleTargets()` with usages such as "any number of target cards with total mana value X or less".
+     * Adds a selection hint for the total value of the targets already chosen.
+     * @param target          this
+     * @param possibleTargets super.possibleTargets()
+     * @param valueMapper     e.g. MageObject::getManaValue or m -> m.getPower().getValue()
+     * @param maxValue        the maximum total value of the parameter
+     * @return the set of possible targets that don't exceed the maximum total value.
+     */
+    public static Set<UUID> checkPossibleTargetsTotalValueLimit(Target target, Set<UUID> possibleTargets,
+                                                                ToIntFunction<MageObject> valueMapper, int maxValue, Game game) {
+        int selectedValue = target.getTargets().stream()
+                .map(game::getObject)
+                .filter(Objects::nonNull)
+                .mapToInt(valueMapper)
+                .sum();
+        target.withChooseHint("selected total value " + selectedValue);
```

Comment:
Do not change inner data in read only methods

Reply:
What's wrong with this usage? It's practical for improved UX, similar to the message in CrewCost, and analogous to the code that was in March from the Tomb before this rework. The choose hint is just a string for display, it doesn't affect any logic.

Reply:
Cause it's a standard code contract all around in software industry. Xmage has only few places with allowed data modification like check trigger (and I don't like it too). Do no spread it in other places.

Must override target's getMessage instead as all other cards does for same task (for adding additional info in target dialog).

Reply:
I agree with the general idea, but I don't really agree with you for this application. But I adjusted this PR per your recommendation, even though it requires more boilerplate code.